app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,585 @@
+        if (colorTint != NO_ID) {
+            ImageViewCompat.setImageTintList(
+                iconImageView,
+                AppCompatResources.getColorStateList(context, colorTint)
+            )
+        }
+    }
+
+    fun setContent(collapsibleContent: View) {
+        containerFrameLayout.addView(collapsibleContent)
+
+        val displayMetrics = DisplayMetrics()
+        (context as? AppCompatActivity)?.windowManager?.defaultDisplay?.getMetrics(displayMetrics)
+            ?.let {
+                containerFrameLayout.measure(
+                    displayMetrics.widthPixels,
+                    displayMetrics.heightPixels
+                )
+                requirementsHeight = containerFrameLayout.measuredHeight
+            }
+
+        if (elementsAreCollapsed) {
+            updateContainerViewHeight()
+            rotateRowIndicator()
+            callOnCollapse(elementsAreCollapsed)
+        }
+
+        updateFooterText(elementsAreCollapsed)
+
+        setOnClickListener { collapseElements(requirementsHeight) }
+    }
+
+    fun seeArrowIndicator(isVisible: Boolean) {
+        footerIndicatorImageView.visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+    }
+
+    fun getContent(): View? {
+        return if (containerFrameLayout.childCount > ZERO) {
+            containerFrameLayout.getChildAt(ZERO)
+        } else {
+            null
+        }
+    }
+
+    fun setTitle(sectionTitle: String?) {
+        sectionTitleTextView.text = sectionTitle
+    }
+
+    fun getTitle(): String? {
+        return sectionTitleTextView.text?.toString()
+    }
+
+    fun setSubtitle(sectionSubtitle: String?) {
+        sectionSubtitleTextView.text = sectionSubtitle
+    }
+
+    fun getSubtitle(): String? {
+        return sectionSubtitleTextView.text?.toString()
+    }
+
+    fun setHiddenFooterText(hiddenFooterText: String?) {
+        textHiddenState = hiddenFooterText
+        updateFooterText(elementsAreCollapsed)
+    }
+
+    fun getHiddenFooterText(): String? {
+        return textHiddenState
+    }
+
+    fun setShowFooterText(showFooterText: String?) {
+        textShowState = showFooterText
+        updateFooterText(elementsAreCollapsed)
+    }
+
+    fun getShowFooterText(): String? {
+        return textShowState
+    }
+
+    fun setFooterTextColor(footerTextColor: Int?) {
+        if (footerTextColor != null && footerTextColor != NO_ID) {
+            footerTextTextView.setTextColor(footerTextColor)
+        }
+    }
+
+    fun setOnCollapseListener(collapseListener: OnCollapseListener?) {
+        this.collapseListener = collapseListener
+    }
+
+    fun startCollapsed() {
+        elementsAreCollapsed = true
+    }
+
+    fun collapse() {
+        if (!elementsAreCollapsed) {
+            collapseElements(requirementsHeight)
+        }
+    }
+
+    interface OnCollapseListener {
+
+        fun onCollapse(isCollapsed: Boolean)
+    }
+
+    companion object {
+
+        fun Int.zeroOrGreaterThanZero(): Int {
```

Comment:
why if this is a method is not the name is not a verb, and why this a global extension

Reply:
It's not a global function, it's internal. Fixed!